atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/AbstractCassandraKeyValueServiceTransactionIntegrationTest.java

```diff
@@ -15,95 +15,48 @@
  */
 package com.palantir.atlasdb.keyvalue.cassandra;
 
-import com.google.common.base.Suppliers;
 import com.google.common.util.concurrent.Futures;
 import com.palantir.atlasdb.AtlasDbConstants;
 import com.palantir.atlasdb.containers.CassandraResource;
 import com.palantir.atlasdb.keyvalue.api.KeyAlreadyExistsException;
-import com.palantir.atlasdb.keyvalue.api.KeyValueService;
 import com.palantir.atlasdb.transaction.api.Transaction;
-import com.palantir.atlasdb.transaction.impl.AbstractTransactionTest;
-import com.palantir.atlasdb.transaction.impl.GetAsyncDelegate;
-import com.palantir.atlasdb.transaction.impl.TransactionConstants;
+import com.palantir.atlasdb.transaction.impl.AbstractTransactionTestV2;
 import com.palantir.atlasdb.transaction.impl.TransactionSchemaVersionEnforcement;
 import com.palantir.atlasdb.transaction.impl.TransactionTables;
 import com.palantir.atlasdb.transaction.service.SimpleTransactionService;
 import com.palantir.atlasdb.transaction.service.TransactionService;
 import com.palantir.atlasdb.transaction.service.WriteBatchingTransactionService;
-import com.palantir.flake.FlakeRetryingRule;
-import com.palantir.flake.ShouldRetry;
-import java.util.Arrays;
-import java.util.Collection;
+import com.palantir.flake.FlakeRetryTest;
 import java.util.List;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
-import java.util.function.Supplier;
 import java.util.function.UnaryOperator;
 import java.util.stream.Collectors;
 import java.util.stream.LongStream;
-import org.junit.Before;
-import org.junit.ClassRule;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TestRule;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-@ShouldRetry // The first test can fail with a TException: No host tried was able to create the keyspace requested.
-@RunWith(Parameterized.class)
-public class CassandraKeyValueServiceTransactionIntegrationTest extends AbstractTransactionTest {
-    private static final String SYNC_TRANSACTIONS_1 = "sync, transactions1";
-    private static final String ASYNC_TRANSACTIONS_2 = "async, transactions2";
-    private static final String ASYNC_TRANSACTIONS_3 = "async, transactions3";
-    private static final Supplier<KeyValueService> KVS_SUPPLIER =
-            Suppliers.memoize(CassandraKeyValueServiceTransactionIntegrationTest::createAndRegisterKeyValueService);
```

Comment:
This part is tricky. We used to create our own `KVS_SUPPLIER` with memoization and register it to `CassandraResource` as `kvs`. However, all these steps were unnecessary since `CassandraResource` automatically creates the same `KVS_SUPPLIER`, memoizes it, and registers it as `kvs`. As far as I can see, this logic was added before `CassandraResource` supported these steps, so it should be safe to remove this logic. 

On the other hand, I noticed that this piece of code was failing in JUnit5 but not in JUnit4, which does not make sense because there is not much difference between the two. As soon as I removed the memoize function here, the tests started passing again. I investigated for a while but couldn't find a reason why the inner memoization (given that `CassandraResource` also wraps it with outer memoization) causes a problem. Since everything is fine now, and the tests are safe and sound, I don't find further debugging necessary. But this is something to think about.

Reply:
Note to the reviewer: I would be glad if you can double check if my reasoning is correct here.